pyteal/ast/assert_.py

```diff
@@ -23,16 +26,25 @@ def __init__(self, cond: Expr, *additional_conds: Expr) -> None:
         require_type(cond, TealType.uint64)
         for cond_single in additional_conds:
             require_type(cond_single, TealType.uint64)
+
+        self.comment = comment
         self.cond = [cond] + list(additional_conds)
 
     def __teal__(self, options: "CompileOptions"):
         if len(self.cond) > 1:
-            asserts = [Assert(cond) for cond in self.cond]
+            asserts: list[Expr] = []
+            for cond in self.cond:
+                asrt = Assert(cond, comment=self.comment)
```

Comment:
I don't love that this duplicate the comment for every child condition, but I'm not sure what would be better

Reply:
Agree, its weird. This seemed like the best compromise and the author can always refactor to individual Asserts.

Reply:
Ok, I'm happy enough with this at the moment to approve. We can revisit the behavior in the future if users complain.